In [48]:
import numpy as np
import pandas as pd
import pyodbc
import sqlite3
import math
import seaborn as sns
from sklearn.model_selection import train_test_split
from imblearn.over_sampling import RandomOverSampler 
from scipy.stats import rv_discrete
from scipy.stats import t, binom, chi2, f
from scipy.stats import ttest_ind

import importlib
import funciones_eda
importlib.reload(funciones_eda)
from funciones_eda import *

import importlib
import funciones_m
importlib.reload(funciones_m)
from funciones_m import *

import importlib
import funciones_distribucion
importlib.reload(funciones_distribucion)
from funciones_distribucion import *

import importlib
import funciones_pruebas_significativas
importlib.reload(funciones_pruebas_significativas)
from funciones_pruebas_significativas import *

In [49]:
url = 'df_filtrado.csv'
df = pd.read_csv(url, index_col=None)
df.drop('index', axis=1, inplace=True)

### Experimentos estadísticos y pruebas significativas

In [50]:
Prueba_mannwhitneyu(df,'TIPOL_TRANS_NAME','QTY','Minimercado Independ','Superete Independ')

Tasa de venta Grupo A Minimercado Independ: 196.34% 
Tasa de venta Grupo B Superete Independ: 526.27%
Estadístico U: 89790622.0
Valor p: 0.0
Se rechaza la hipótesis nula: las distribuciones son diferentes.


In [51]:
Prueba_mannwhitneyu_bidireccional(df,'TIPOL_TRANS_NAME','QTY','Minimercado Independ','Superete Independ')

Tasa de venta Grupo A Minimercado Independ: 196.34% 
Tasa de venta Grupo B Superete Independ: 526.27%
Estadístico U: 89790622.0
Valor p: 0.0
Conclusión: Se rechaza la hipótesis nula: el grupo 1 tiene valores significativamente menores que el grupo 2.


Grupo A (Minimercado Independ): Tiene una  promedio compra del 196.34%.
Grupo B (Superete Independ): Tiene una promedio de compra significativamente más alta, del 526.27%.

Esto indica que, en promedio, los clientes del grupo "Superete Independ" compran aproximadamente 5.26 veces la cantidad base, mientras que los del grupo "Minimercado Independ" compran alrededor de 1.96 veces la cantidad base.


**prueba de hipotesis**

La prueba muestra una diferencia estadísticamente significativa entre los grupos.
Esto significa que el comportamiento de compra (medido por QTY) es significativamente diferente entre los clientes de los grupos "Minimercado Independ" y "Superete Independ."

Los clientes del grupo "Superete Independ" tienden a realizar compras más grandes o frecuentes en comparación con "Minimercado Independ".